### ISSUE TITLE
Filter ringing from dropped data points

### ISSUE THREAD

**Issue by jdpigeon** [alexcastillo/eeg-pipes#9]
One unfortunate side effect of bluetooth EEG is the occasional dropped data point. Standard procedure in EEG analysis is to mark dropped samples as data is collected and make sure those time points are excluded from analysis.

When using linear filters in real time, though, dropped data points can cause conspicuous ringing artifacts. 

Unfortunately, the fili library doesn't not handle NaN or null values well. If any NaNs are passed through a filter function, the entire array will return as NaNs. null is treated as 0, which is undesirable as well.

It's up to users of this library to decide how they want to deal with dropped data. Smoothing around dropped data points by averaging points before and after is a good solution. Maybe we could write an operator to handle that.

**Comment by alexcastillo** [comment]
I like the proposal and solution!

**Comment by jdpigeon** [comment]
I rewrote some of the filter functions to handle dropped data in this commit here: https://github.com/jdpigeon/eeg-pipes/commit/f55798bfc2267dc20132b1986b5c387d3121e1cf

What I did is map through incoming data, replacing NaNs, nulls, or undefineds with interpolated fake values before the data goes through the filter function. Then, after the filter, insert the NaNs, nulls, or undefineds where they were in the data.

My implementation might not be the best as I've noticed it takes a little bit longer to run the filter operations with this new 'safe' approach (before <1ms now  ~1.5ms on ~66 samples). I'll do some testing on real hardware soon to see if this significant at all
